modules/aws/lb/main.tf

```diff
@@ -0,0 +1,272 @@
+locals {
+  is_application = var.load_balancer_type == "application"
+  is_network     = var.load_balancer_type == "network"
+}
+
+# Load Balancer
+resource "aws_lb" "load_balancer" {
+  # Common settings
+  name                       = var.name
+  internal                   = var.internal
+  load_balancer_type         = var.load_balancer_type
+  subnets                    = var.subnets
+  enable_deletion_protection = var.enable_deletion_protection
+  customer_owned_ipv4_pool   = var.customer_owned_ipv4_pool
+  ip_address_type            = var.ip_address_type
+
+  # Application Load Balancer specific settings
+  security_groups            = local.is_application ? var.security_groups : null
+  desync_mitigation_mode     = local.is_application ? var.desync_mitigation_mode : null
+  idle_timeout               = local.is_application ? var.idle_timeout : null
+  drop_invalid_header_fields = local.is_application ? var.drop_invalid_header_fields : null
+  enable_http2               = local.is_application ? var.enable_http2 : null
+  enable_waf_fail_open       = local.is_application ? var.enable_waf_fail_open : null
+
+  # Network Load Balancer specific settings
+  enable_cross_zone_load_balancing = local.is_network ? var.enable_cross_zone_load_balancing : null
+
+  dynamic "access_logs" {
+    for_each = var.access_logs != null ? [var.access_logs] : []
```

Comment:
Change this to remove the brackets around var.access_logs. 

If we need each type to be a set then we should set that requirement within the variable rather than turning the variable into a set this way. If instead, you do want to take the input of a single object and turn it into a set a better more prescriptive way would be to use 
`toset()`. This tells the user that the variable will turn into a set.

Example:
```
for_each = var.access_logs != null ? var.access_logs : []
```
Or:
```
for_each = var.access_logs != null ? toset(var.access_logs) : []
```

Reply:
I ended up doing
`for_each = var.access_logs != null ? { create = var.access_logs } : {}`

Does this work? var is now map of object.